doc/source/io.rst

```diff
@@ -4752,6 +4752,57 @@ default ``Text`` type for string columns:
     Because of this, reading the database table back in does **not** generate
     a categorical.
 
+.. _io.sql.method:
+
+Insertion Method
+++++++++++++++++
+
+.. versionadded:: 0.24.0
+
+The parameter ``method`` controls the SQL insertion clause used.
+
+Possible values are:
+
+- `'default'`: Uses standard SQL `INSERT` clause (one per row).
+- `'multi'`: Pass multiple values in a single `INSERT` clause.
+  It uses a **special** SQL syntax not supported by all backends.
+  This usually provides better performance for Analytic databases
+  like *Presto* and *Redshit*, but has worse performance for
+  traditional SQL backend if the table contains many columns.
+  For more information check SQLAlchemy `documention
+  <http://docs.sqlalchemy.org/en/latest/core/dml.html?highlight=multivalues#sqlalchemy.sql.expression.Insert.values.params.*args>`__.
+- callable: with signature `(pd_table, conn, keys, data_iter)`.
+  This can be used to implement more performant insertion based on
+  specific backend dialect features.
+  I.e. using *Postgresql* `COPY clause
+  <https://www.postgresql.org/docs/current/static/sql-copy.html>`__.
+
+Example of callable for Postgresql *COPY*::
+
+  # Alternative to_sql() *method* for DBs that support COPY FROM
+  import csv
+  from io import StringIO
+
+  def psql_insert_copy(table, conn, keys, data_iter):
+      # gets a DBAPI connection that can provide a cursor
+      dbapi_conn = conn.connection
+      with dbapi_conn.cursor() as cur:
+          s_buf = StringIO()
+          writer = csv.writer(s_buf)
+          writer.writerows(data_iter)
+          s_buf.seek(0)
+
+          columns = ', '.join('"{}"'.format(k) for k in keys)
+          if table.schema:
+              table_name = '{}.{}'.format(table.schema, table.name)
+          else:
+              table_name = table.name
+
+          sql = 'COPY {} ({}) FROM STDIN WITH CSV'.format(
+              table_name, columns)
```

Comment:
I realise this is just an example, but isn't this piece vulnerable to SQL injection? the[`psycopg2.sql`](http://initd.org/psycopg/docs/sql.html) module has an [`Identifier`](http://initd.org/psycopg/docs/sql.html#psycopg2.sql.Identifier) object designed to handle SQL identifiers appropriately. See also [Example 41-1. Quoting Values In Dynamic Queries](https://www.postgresql.org/docs/9.6/static/plpgsql-statements.html#PLPGSQL-QUOTE-LITERAL-EXAMPLE).

Reply:
@dahlbaek no expert on this, but if you could provide the code that you would replace it with to make it more robust, that would certainly be welcome!